### PR TITLE
Pagination logic refactoring

### PR DESCRIPTION
**Pull request by Grasmus**

### CODE REVIEW

File: API/API.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
-    <UserSecretsId></UserSecretsId>
+    <UserSecretsId>61e8bdb4-170e-4ea9-9e68-3b0ac08540bf</UserSecretsId>
```
Comment (Eugene-pas) [inline API/API.csproj]: You forgot to delete the key

File: Core/Services/UserService.cs [hunk 2]
```diff
@@ -101,20 +107,26 @@ public async Task<UserFullNameDTO> GetUserFullNameAsync(string userId)
...
             }
 
-            return new PaginatedList<UserDTO>(
-                _mapper.Map<List<UserDTO>>(paginatedUserList.Items), 
-                paginationFilter.PageNumber, 
-                paginatedUserList.TotalPages, 
-                paginatedUserList.TotalItems);
+            var userList = await _userRepository.ListAsync(new UserSpecification.GetByIds(userIds));
+
+            return PaginatedList<UserDTO>.Evaluate(_mapper.Map<List<UserDTO>>(userList), paginationFilter, userRolesCount);
```
Comment (Eugene-pas) [inline Core/Services/UserService.cs]: This string is longer than 100 characters, move it to a new line.

File: Core/Services/UserService.cs [hunk 3]
```diff
@@ -101,20 +107,26 @@ public async Task<UserFullNameDTO> GetUserFullNameAsync(string userId)
...
-                return null;
+                userIds.Add(userRole.UserId);
             }
 
-            return new PaginatedList<UserDTO>(
-                _mapper.Map<List<UserDTO>>(paginatedUserList.Items), 
-                paginationFilter.PageNumber, 
-                paginatedUserList.TotalPages, 
-                paginatedUserList.TotalItems);
+            var userList = await _userRepository.ListAsync(new UserSpecification.GetByIds(userIds));
```
Comment (Eugene-pas) [inline Core/Services/UserService.cs]: This string is longer than 100 characters, move it to a new line.

File: Core/Specifications/UserSpecification.cs [hunk 4]
```diff
@@ -12,12 +13,21 @@ public GetByEmail(string email)
                 Query.Where(u => u.Email == email);
             }
         }
+
         internal class GetByConfirmationToken : Specification<User>, ISingleResultSpecification<User>
```
Comment (Eugene-pas) [inline Core/Specifications/UserSpecification.cs]: This string is longer than 100 characters, move it to a new line.

**Review by Eugene-pas — APPROVED** [review]

**Review by AndriiChepeliuk — APPROVED** [review]

**Pull request merged**